src/components/dialog/dialog/dialog-base-element.ts

```diff
@@ -0,0 +1,157 @@
+import { SbbOverlayBaseElement } from '../../core/base-elements.js';
+import { SbbLanguageController } from '../../core/controllers.js';
+import { hostContext, SbbScrollHandler } from '../../core/dom.js';
+import { EventEmitter } from '../../core/eventing.js';
+import { i18nDialog } from '../../core/i18n.js';
+import type { SbbDialogCloseEventDetails } from '../../core/interfaces.js';
+import { SbbNegativeMixin } from '../../core/mixins.js';
+import { applyInertMechanism, removeInertMechanism } from '../../core/overlay.js';
+import type { SbbScreenReaderOnlyElement } from '../../screen-reader-only.js';
+
+// A global collection of existing dialogs
+export const dialogRefs: SbbDialogBaseElement[] = [];
+
+export abstract class SbbDialogBaseElement extends SbbNegativeMixin(SbbOverlayBaseElement) {
+  /** This will be forwarded as aria-label to the relevant nested element to describe the purpose of the dialog. */
+  @property({ attribute: 'accessibility-label' }) public accessibilityLabel: string | undefined;
+
+  /** Emits whenever the `sbb-dialog` is closed. */
+  protected override didClose: EventEmitter<SbbDialogCloseEventDetails> = new EventEmitter(
+    this,
+    SbbDialogBaseElement.events.didClose,
+  );
+
+  // The last element which had focus before the dialog was opened.
+  protected lastFocusedElement?: HTMLElement;
+  protected dialogCloseElement?: HTMLElement;
+  protected dialogController!: AbortController;
+  protected openDialogController!: AbortController;
+  protected focusHandler = new SbbFocusHandler();
+  protected scrollHandler = new SbbScrollHandler();
+  protected returnValue: any;
+  protected ariaLiveRefToggle = false;
+  protected ariaLiveRef!: SbbScreenReaderOnlyElement;
+  protected language = new SbbLanguageController(this);
+
+  protected abstract onDialogAnimationEnd(event: AnimationEvent): void;
+  protected abstract setDialogFocus(): void;
+  protected abstract closeAttribute: string;
+
+  /** Closes the component. */
+  public close(result?: any, target?: HTMLElement): any {
+    if (this.state !== 'opened') {
+      return;
+    }
+
+    this.returnValue = result;
+    this.dialogCloseElement = target;
+    const eventData: SbbDialogCloseEventDetails = {
+      returnValue: this.returnValue,
+      closeTarget: this.dialogCloseElement,
+    };
+
+    if (!this.willClose.emit(eventData)) {
+      return;
+    }
+    this.state = 'closing';
+    this.removeAriaLiveRefContent();
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    this.dialogController?.abort();
+    this.dialogController = new AbortController();
+
+    if (this.state === 'opened') {
+      applyInertMechanism(this);
+    }
+  }
+
+  protected override firstUpdated(changedProperties: PropertyValues<this>): void {
+    this.ariaLiveRef =
+      this.shadowRoot!.querySelector<SbbScreenReaderOnlyElement>('sbb-screen-reader-only')!;
+
+    super.firstUpdated(changedProperties);
```

Comment:
Is it necessary to have super call at the end?

Reply:
If i remove it, the `ESLint: Method needs super call.(lyne/needs-super-call-rule)` error appears.